kernel/sched.c

```diff
@@ -233,7 +233,8 @@ static inline int sliceable(struct k_thread *t)
 	return _is_preempt(t)
 		&& !_is_prio_higher(t->base.prio, slice_max_prio)
 		&& !_is_idle(t)
-		&& !_is_thread_timeout_active(t);
+		&& !_is_thread_timeout_active(t)
+		&& _is_thread_queued(t);
```

Comment:
This sounds wrong.  A thread "in the process of pending itself" surely shouldn't be interrupted by a timer interrupt.  The time period between the list removal and the _Swap() should be under a lock, and I'm pretty sure I chased that last night and convinced myself it was true.

I buy that this works to work around the bug, but I don't think it's a root cause.  We need to find out why we're in this state (i.e. point to a line of code and say "this is where it can be interrupted while unqueued") and then make sure that's locked or fix the design bug that prevents it.

Reply:
What appears to be happening is one of the worker threads in the test completes and exits, which causes k_thread_abort() to be invoked on it while it is still `_current`.  This does occur under lock, terminated by a call to __swap().

__swap() twiddles some state, then releases the IRQ lock and relies on a triggered PendSV handler to complete the context switch and update `_current`.

Before that can happen, a higher priority timer ISR fires which initiates a timeslice.  The aborted thread is still in `_current`, which is no longer in any queue, and if the slice is taken a double-remove is issued.  Prior to this PR that didn't cause any visible harm; now it does.

The proposed patch works because it blocks taking the slice from an unqueued `_current`.  From what I can tell this is a pre-existing bug, revealed but not introduced by this PR, so the workaround should be acceptable within the scope of this PR.

I've changed the commit message of the other commit as requested.